docs/components.rst

```diff
@@ -21,6 +21,93 @@ Each handler is associated to a service and is used to process different events
+`secure-data-proxy` service, this service handles the permissions of those files. If a file does not have a
+corresponding route on the `secure-data-proxy` service, it will use the closest parent permissions. Note that the route
+resources found under the `secure-data-proxy` service must match exactly a path on the filesystem, starting with the
+directory name ``wpsoutputs``, and following with the desired children directories/file names. If no `secure-data-proxy`
+service is found, the user files are assumed to be fully available with read and write permissions for the user. Note
+that if the file does not have any read or write permissions, the hardlink will not be available in the user's
+workspace.
+
+Note that different design choices were made to respect the constraints of the file system and to prevent the user from
+accessing forbidden data:
+
+- To avoid having to copy every file from the wps outputs data directory to the corresponding user workpaces, the usage
+  of links was chosen.
+- The usage of hardlinks was chosen instead of symlinks. A reason is that using symlinks would require mounting the
+  wps outputs data directory on the user `JupyterLab` instance. This is because symlinks require access to their source
+  files, contrary to hardlinks which can be mounted individually without access to source files. Mounting the
+  whole wps outputs data directory would give the user access to all the data. Even if the data is
+  not made available on the `JupyterLab` browser via the UI, the data would still be accessible via the terminal found
+  on `JupyterLab`. Using hardlinks lets us mount only the public directory instead of the whole wps outputs directory,
+  which contains a mix of public and user data.
+- Note that changing permissions to control user access was not an option since symlinks and hardlinks must have the
+  same permissions as their original source file.
+  Therefore, it not possible for example to give access to a user and prevent the access to another user for a specific
+  file, even if using hardlinks or symlinks.
+- It is possible to add anonymous volumes to hide some parts of the wps outputs data directory, but it was easier
+  and safer to go with the hardlinks option to prevent errors which could accidentally give access to other users' data.
+  For example, if a bird folder was created while a user `JupyterLab` instance was running, it would require both the
+  instance and `JupyterHub` to be restarted in order to generate the new required anonymous volume to hide the user data
+  from that new bird. Without a restart of the instance, the user could potentially have access to some of the new user
+  data.
+
+In conclusion, the best option was to use hardlinks, which do not require access to the original source file, to create
+separate user and public data access points, and volume mounting to control which locations are made available to the
+user.
+
```

Comment:
Because all these considerations come from multiple lengthy discussion/reflections, I think it would be worth adding some `.. seealso::` block with references to the Jira issue and other relevant Confluence pages linked in it for traceability. They won't be accessible externally, but they can help up find important decisions quickly if we ever need to come back on this.